### PR TITLE
new: adding `region-table` plugin

### LINKED ISSUE #492: [Feature]: I want to see a table listing the functions of each region.
### Description

Linode is currently creating many new regions.

Since functional differences have been created, it is necessary to briefly review the functionality of the regions.

Could you please add a command option so that we can check it in a table or csv like the following.

```
linode-cli regions ls --json | jq '[.[] | del(.resolvers) | del(.status)]' | python3 test.py

+--------------+----------------+---------+---------+-------------+---------------+---------------+---------------+----------------+------------+----------------+-------+-------------------+
|      ID      |     Label      | Country | Linodes | GPU Linodes | Premium Plans | NodeBalancers | Block Storage | Object Storage | Kubernetes | Cloud Firewall | Vlans | Managed Databases |
+--------------+----------------+---------+---------+-------------+---------------+---------------+---------------+----------------+------------+----------------+-------+-------------------+
|   ap-west    |   Mumbai, IN   |    in   |    ✔    |      ✔      |       -       |       ✔       |       ✔       |       -        |     ✔      |       ✔        |   ✔   |         ✔         |
|  ca-central  |  Toronto, CA   |    ca   |    ✔    |      -      |       -       |       ✔       |       ✔       |       -        |     ✔      |       ✔        |   ✔   |         ✔         |
| ap-southeast |   Sydney, AU   |    au   |    ✔    |      -      |       -       |       ✔       |       ✔       |       -        |     ✔      |       ✔        |   ✔   |         ✔         |
|    us-iad    | Washington, DC |    us   |    ✔    |      -      |       ✔       |       ✔       |       ✔       |       ✔        |     ✔      |       ✔        |   ✔   |         ✔         |
|    us-ord    |  Chicago, IL   |    us   |    ✔    |      -      |       ✔       |       ✔       |       ✔       |       ✔        |     ✔      |       ✔        |   ✔   |         ✔         |
|    fr-par    |   Paris, FR    |    fr   |    ✔    |      -      |       ✔       |       ✔       |       ✔       |       ✔        |     ✔      |       ✔        |   ✔   |         ✔         |
|  us-central  |   Dallas, TX   |    us   |    ✔    |      -      |       -       |       ✔       |       ✔       |       -        |     ✔      |       ✔        |   -   |         ✔         |
|   us-west    |  Fremont, CA   |    us   |    ✔    |      -      |       -       |       ✔       |       ✔       |       -        |     ✔      |       ✔        |   -   |         ✔         |
| us-southeast |  Atlanta, GA   |    us   |    ✔    |      ✔      |       -       |       ✔       |       ✔       |       ✔        |     ✔      |       ✔        |   ✔   |         ✔         |
|   us-east    |   Newark, NJ   |    us   |    ✔    |      ✔      |       -       |       ✔       |       ✔       |       ✔        |     ✔      |       ✔        |   ✔   |         ✔         |
|   eu-west    |   London, UK   |    uk   |    ✔    |      -      |       -       |       ✔       |       ✔       |       -        |     ✔      |       ✔        |   ✔   |         ✔         |
|   ap-south   | Singapore, SG  |    sg   |    ✔    |      ✔      |       -       |       ✔       |       ✔       |       ✔        |     ✔      |       ✔        |   ✔   |         ✔         |
|  eu-central  | Frankfurt, DE  |    de   |    ✔    |      ✔      |       -       |       ✔       |       ✔       |       ✔        |     ✔      |       ✔        |   ✔   |         ✔         |
| ap-northeast |   Tokyo, JP    |    jp   |    ✔    |      -      |       -       |       ✔       |       ✔       |       -        |     ✔      |       ✔        |   -   |         ✔         |
+--------------+----------------+---------+---------+-------------+---------------+---------------+---------------+----------------+------------+----------------+-------+-------------------+
```

```
test.py

import sys
import json
from prettytable import PrettyTable

json_data = sys.stdin.read()

data = json.loads(json_data)

capabilities = ["Linodes", "GPU Linodes", "Premium Plans", "NodeBalancers", "Block Storage", "Object Storage", "Kubernetes", "Cloud Firewall", "Vlans", "Managed Databases"]

# Create a PrettyTable instance
x = PrettyTable()

# Add columns
x.field_names = ["ID", "Label", "Country"] + capabilities

# Center align all columns
x.align = "c"

# Add rows
for d in data:
    row = [d["id"], d["label"], d["country"]] + ["✔" if c in d["capabilities"] else "-" for c in capabilities]
    x.add_row(row)

print(x)
```

### Potential Usage Example

_No response_

### PR DESCRIPTION
**Pull request by jriddle-linode**
## 📝 Description

**What does this PR do and why is this change necessary?**

Plugin adding `region-table` command to the CLI.
```bash
lin region-table
```

## ✔️ How to Test

**How do I run the relevant unit/integration tests?**

`make INTEGRATION_TEST_PATH=regions testint`

## 📷 Preview

image

resolves #492

### CODE REVIEW

File: linodecli/plugins/region-table.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        "GPU Linodes",
+        "Premium Plans",
+        "NodeBalancers",
+        "Block Storage",
+        "Object Storage",
+        "Kubernetes",
+        "Cloud Firewall",
+        "Vlans",
+        "Managed Databases",
+    ]
```
Comment (lgarber-akamai) [inline linodecli/plugins/region-table.py:38]: optional: Maybe we could use tuples to couple the short-form and long-form capability names
```python
capabilities = [
    ("Linodes", "Linodes"),
    ("GPU Linodes", "GPU"),
    ("Premium Plans", "Premium"),
    ("NodeBalancers", "NB"),
    ("Block Stoarge", "Blocks"),
    ("Object Storage", "Objects"),
    ("Kubernetes", "K8s"),
    ("Firewalls", "FW"),
    ("Vlans", "Vlans"),
    ("Managed Databases", "DB"),
]
```

File: linodecli/plugins/region-table.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
+"""
+The region-table plugin displays a table output
+for the capabilities of each region.
+"""
+import sys
+
+from prettytable import PrettyTable
```
Comment (lgarber-akamai) [inline linodecli/plugins/region-table.py:7]: Could we use rich tables here so we can consume the existing rich dep?

Comment (lgarber-akamai) [inline linodecli/plugins/region-table.py:7]: Could we use rich tables here so we can consume the existing rich dep?

**Review by lgarber-akamai — CHANGES_REQUESTED** [review]

File: linodecli/plugins/region-table.py (hunk 1 above)
Comment (isss802) [reply linodecli/plugins/region-table.py:38]: @lgarber-akamai 
Could you also add Metadata Service to the column?

Comment (lgarber-akamai) [reply linodecli/plugins/region-table.py:38]: cc @jriddle-linode

**Review by amisiorek-akamai — APPROVED** [review]
LGTM!

**Review by yec-akamai — APPROVED** [review]
Nice work! Tests passed locally 🎉

**Review by lgarber-akamai — APPROVED** [review]
Works well in my local testing, nice work!

**Pull request merged**